features/features.go

```diff
@@ -166,10 +169,26 @@ var featuresSupportedCallbacks = map[SnapdFeature]func() (bool, string){
 		}
 		return true, ""
 	},
-	// AppArmorPrompting requires a newer version of snapd with all the
+	// AppArmorPrompting requires AppArmor parser and kernel support for
+	// prompting, as well as a newer version of snapd with all the
 	// prompting components in place. TODO: change this callback once ready.
 	AppArmorPrompting: func() (bool, string) {
+		kernelFeatures, err := apparmor.KernelFeatures()
+		if err != nil {
+			return false, fmt.Sprintf("error checking apparmor kernel features: %v", err)
+		}
+		if !strutil.ListContains(kernelFeatures, "policy:permstable:prompt") {
```

Comment:
You're correct, I made a mistake in the initial commit for this PR, which has since been fixed in the separate `AppArmorPrompting.IsSupported()` PR which correctly checks apparmor features: https://github.com/snapcore/snapd/pull/13821